pkg/activator/handler/concurrency_reporter.go

```diff
@@ -204,3 +200,15 @@ func (cr *ConcurrencyReporter) run(stopCh <-chan struct{}, reportCh <-chan time.
 		}
 	}
 }
+
+// Handler returns a handler that records request coming in/being finished in the stats
+// machinery.
+func (cr *ConcurrencyReporter) Handler(next http.Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		revisionKey := util.RevIDFrom(r.Context())
+		cr.handleEvent(network.ReqEvent{Key: revisionKey, Type: network.ReqIn, Time: time.Now()})
+		defer cr.handleEvent(network.ReqEvent{Key: revisionKey, Type: network.ReqOut, Time: time.Now()})
```

Comment:
I _think_ this is wrong.. the args to defer get evalulated straight away, so both time.Nows are (basically) the same :(. Needs to be
```suggestion
		defer func() { cr.handleEvent(network.ReqEvent{Key: revisionKey, Type: network.ReqOut, Time: time.Now()}) }()
```

Reply:
Great catch! You're absolutely right!